### PR TITLE
Add docs for HTTPRoute timeouts + retries + route metrics

### PR DESCRIPTION
**Pull request by adleong**

### CODE REVIEW

File: linkerd.io/content/2.16/reference/timeouts.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+description = "How Linkerd implements timeouts."
++++
+
+Linkerd can be configured with timeouts to limit the maximum amount of time on
+a request before aborting.
+
+Timeouts are a client-side behavior, and are therefore performed by the
+outbound side of the Linkerd proxy.[^1] Note that if these timeouts are reached,
+the request will not be retried. Retry timeouts can be configured as part of
+[retry configuration](../retries/).
```
Comment (kflynn) [inline linkerd.io/content/2.16/reference/timeouts.md:12]:
```suggestion
outbound side of the Linkerd proxy.[^1] Note that configuring timeouts alone will _not_ automatically retry requests that time out! You'll need to also [configure Linkerd retries](../retries/) if you want that to happen.
```

File: linkerd.io/content/2.16/reference/timeouts.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+## Configuring Timeouts
+
+Timeous are configured by a set of annotations which can be set on a Kubernetes
+Service resource or on a HttpRoute or GrpcRoute which has a Service as a parent.
```
Comment (kflynn) [inline linkerd.io/content/2.16/reference/timeouts.md:17]:
```suggestion
Service resource, or on a HTTPRoute or GRPCRoute which has a Service as a parent.
```

File: linkerd.io/content/2.16/reference/timeouts.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+Service.
+
+{{< warning >}}
+Timeouts configured in this way are **incompatible with ServiceProfiles**. If a
+[ServiceProfile](../../features/service-profiles/) is defined for a Service,
+proxies will use the ServiceProfile timeout configuration and ignore any timeout
+annotations.
+{{< /warning >}}
+
++ `timeout.linkerd.io/request`: The maximum amount of time a full
```
Comment (kflynn) [inline linkerd.io/content/2.16/reference/timeouts.md:30]: I wonder if we should relate this back to ServiceProfile timeouts, or timeouts as specified by an HTTPRoute `spec.timeout` stanza...

**Review by kflynn — CHANGES_REQUESTED** [review]
Overall I think this is really good! I've suggested some edits, but they're pretty minor things, really. 

I wholeheartedly agree that not having Viz makes this kind of documentation much tougher, but I think you've done about as well as could be done there. Thanks!!

File: linkerd.io/content/2.16/reference/timeouts.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+request-response stream is in flight.
++ `timeout.linkerd.io/response`: The maximum amount of time a backend response
+may be in-flight.
```
[the remark this answers is not in the document]
Comment (adleong) [reply linkerd.io/content/2.16/reference/timeouts.md:33]: I don't think those are accurate.  My understanding is that the request timeout is measured from Linkerd sends the request header to when both request and response streams are closed and the response timeout is measured from when Linkerd receives the response header until the response stream is closed.  But I'm not 100% sure and I'm inclined to leave this a bit vague until we can confirm the exact behavior.

File: linkerd.io/content/2.16/tasks/getting-per-route-metrics.md [hunk 5]
```diff
@@ -3,101 +3,22 @@ title = "Getting Per-Route Metrics"
...
-To get per-route metrics, you must first create a
-[service profile](../../features/service-profiles/). Once a service
-profile has been created, Linkerd will add labels to the Prometheus metrics that
-associate a specific request to a specific route.
-
-For a tutorial that shows this functionality off, check out the
+To get per-route metrics, you must create [HTTPRoute] resources. If a route has
+a `parent_ref` which points to a Service resource, Linkerd will generate
+outbound per-route traffic metrics for all HTTP traffic that it sends to that
+Service. If a route has a `parent_ref` which points to a Server resource,
```
[the remark this answers is not in the document]
Comment (adleong) [reply linkerd.io/content/2.16/tasks/getting-per-route-metrics.md:9]: That's correct

File: linkerd.io/content/2.16/features/retries-and-timeouts.md [hunk 6]
```diff
@@ -5,25 +5,14 @@ weight = 3
...
-
-Timeouts can be configured using either the [HTTPRoute] or [ServiceProfile]
-resources. Currently, retries can only be configured using [ServiceProfile]s,
-but support for configuring retries using [HTTPRoutes] will be added in a future
-release. Creating these policy resources will cause the Linkerd proxy to perform
-the appropriate retries or timeouts when calling that service. Retries and
-timeouts are always performed on the *outbound* (client) side.
+Timeouts and retries can be configured using [HTTPRoute], GrpcRoute, or Service
+resources. Retries and timeouts are always performed on the *outbound* (client)
+side.
```
[the remark this answers is not in the document]
Comment (kflynn) [reply linkerd.io/content/2.16/features/retries-and-timeouts.md:13]: Sounds good!

File: linkerd.io/content/2.16/tasks/getting-per-route-metrics.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (kflynn) [reply linkerd.io/content/2.16/tasks/getting-per-route-metrics.md]: Thanks for confirming! I think I'm gonna have to play with this a bit. :sli

File: linkerd.io/content/2.16/reference/timeouts.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (kflynn) [reply linkerd.io/content/2.16/reference/timeouts.md:34]: Leaving it vague sounds good then. 🙂

**Review by kflynn — APPROVED** [review]
Ship it! 🙂

**Pull request merged**